src/core/geometry/qgsgeometry.cpp

```diff
@@ -996,6 +1019,42 @@ Qgis::GeometryOperationResult QgsGeometry::addPart( QgsAbstractGeometry *part, Q
   return QgsGeometryEditUtils::addPart( d->geometry.get(), std::move( p ) );
 }
 
+Qgis::GeometryOperationResult QgsGeometry::addPartV2( QgsAbstractGeometry *part, Qgis::WkbType wkbType )
+{
+  std::unique_ptr< QgsAbstractGeometry > p( part );
+  if ( !d->geometry )
+  {
+    switch ( QgsWkbTypes::flatType( wkbType ) )
```

Comment:
This method should also accept the multi types:

```suggestion
    switch ( QgsWkbTypes::singleType( QgsWkbTypes::flatType( wkbType ) ) )
```